store/db.go

```diff
@@ -49,15 +51,17 @@ func newDBLock(dirPath string) (*dbLock, error) {
 }
 
 func (dl *dbLock) lock() error {
+	dl.Lock()
 	if err := dl.fl.ExclusiveLock(); err != nil {
+		dl.Unlock()
 		return err
 	}
-	dl.Lock()
 	return nil
 }
 
 func (dl *dbLock) unlock() error {
 	if err := dl.fl.Unlock(); err != nil {
+		dl.Unlock()
```

Comment:
> I agree that it's not likely that the fl unlock will fail. Because of that, having a log or panic or store-instance cleanup all seem like plausible paths forwards for this PR. I'm fine with my concern not being fully fixed, since it's out of scope of the other fix here. At the very least though, I'd want a comment.

@euank I added a (quite long) comment trying to explain the problem. In the meantime I removed the mutex unlock and made it panic when flock unlock fails (let's see if and when it'll happen)